lms/djangoapps/grades/scores.py

```diff
@@ -188,16 +195,19 @@ def _get_score_from_persisted_or_latest_block(persisted_block, block, weight):
     the latest block content.
     """
     raw_earned = 0.0
+    attempted = False
 
     if persisted_block:
         raw_possible = persisted_block.raw_possible
     else:
         raw_possible = block.transformer_data[GradesTransformer].max_score
 
     if raw_possible is None:
-        return (raw_earned, raw_possible) + (None, None)
+        weighted_scores = (None, None)
     else:
-        return (raw_earned, raw_possible) + weighted_score(raw_earned, raw_possible, weight)
+        weighted_scores = weighted_score(raw_earned, raw_possible, weight)
```

Comment:
The conditional is currently there because of the defensive code that we needed to add for scorable blocks that didn't define a `max_score` value.  I have updated the code with a TODO statement and a JIRA ticket to address the issue and remove this conditional altogether.

In the meantime, I'd like to keep the defensive code within this higher function rather than pushing it down to the lower-level function that is used by others.  In fact, the lower-level function (weighted_score) asserts that the `raw_possible` value is not `None`.  This is because all other code paths should not present a Null value for `raw_possible`.